portfolio/src/main/webapp/script.js

```diff
@@ -12,14 +12,128 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var map;
+var locationsDict = new Object();
+
+/** Calls all necessary functions to be called onLoad */
+function initialize() {
+  loadComments();
+  createMap();
+}
+
+/** Creates a map with all components and adds it to the page. */
+function createMap() {
+  map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: 41.8996099, lng: -87.6232902, h: -7, p: 22 },
+    zoom: 13,
+  });
+
+  setupMarkers();
+  changeView(locationsDict["Bean"]);
+}
+
+/** All setup code for markers. */
+function setupMarkers() {
+  locationsDict["Bean"] = {
+    m: null,
+    lat: 41.8826099,
+    lng: -87.6232902,
+    h: -7,
+    p: 22,
+    is:
+      '<p style="margin-top:-3px; margin-bottom:-3px" class="field-text">Cloud Gate</p>',
+    iw: null,
+  };
+  locationsDict["Skyline"] = {
+    m: null,
+    lat: 41.9147471,
+    lng: -87.6209924,
+    h: 185,
+    p: 17,
+    is:
+      '<p style="margin-top:-3px; margin-bottom:-3px" class="field-text">Photography Point</p>',
+    iw: null,
+  };
+  locationsDict["Riverwalk"] = {
+    m: null,
+    lat: 41.8877563,
+    lng: -87.6273952,
+    h: 125,
+    p: 30,
+    is:
+      '<p style="margin-top:-3px; margin-bottom:-3px" class="field-text">Chicago Riverwalk</p>',
+    iw: null,
+  };
+
+  for (var location in locationsDict) {
+    locationsDict[location].m = createMarker(locationsDict[location]);
+    locationsDict[location].iw = new google.maps.InfoWindow({
```

Comment:
I may be misunderstanding something, but does "m" and "iw" need to be fields of the object. It seems like it can be  stored in a local variable.